### PR TITLE
WIP To discuss changes

### PR DESCRIPTION
**Pull request by nitishkrishna**

### CODE REVIEW

File: src/config/fabric-ansible/ansible-playbooks/common/plugin_ironic.py [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+        self.cc_node.create_cc_node(cc_node_payload)
+
+
+    def register_nodes(self, node_ipmi_details):
+        registered_nodes_list = []
+        for node in node_ipmi_details:
+            ironic_node = {}
+            ironic_node['driver'] = 'pxe_ipmitool'
+            ironic_node['driver_info'] = {}
+            ironic_node['driver_info']['ipmi_address'] = node['host']
```
Comment (nitishkrishna) [inline src/config/fabric-ansible/ansible-playbooks/common/plugin_ironic.py]: change host to address

File: src/config/fabric-ansible/ansible-playbooks/filter_plugins/discover_server.py [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+        return token
+    
+    def get_cc_token(self,cc_ip, username, password):
+        keystone_url = "https://" + cc_ip + ":9091/keystone/v3/auth/tokens?nocatalog"
+        print "GET CC TOKEN : openstack-url ", keystone_url
+        token = self.get_keystone_token(keystone_url, username, password, verify=False)
+
+        return token
+
+    def check_nodes_with_cc(self, ipmi_nodes_detail, cc_node_details):
```
Comment (nitishkrishna) [inline src/config/fabric-ansible/ansible-playbooks/filter_plugins/discover_server.py]: move to CCNode class

File: src/config/fabric-ansible/ansible-playbooks/filter_plugins/discover_server.py [hunk 3]
```diff
@@ -0,0 +1,396 @@
...
+        #}
+        #cc_auth = {
+        #'auth_host': "10.87.69.79",
+        #'username': "admin",
+        #'password': "contrail123"
+        #}
+        #introspection_flag = False 
+        #ironic_node = ImportIronicNodes(auth_args=my_auth_args,
+        #cc_auth_args=cc_auth )
+        #print "CC-NODE:" , ironic_node.cc_node.auth_token
```
Comment (nitishkrishna) [inline src/config/fabric-ansible/ansible-playbooks/filter_plugins/discover_server.py]: remove this section of commented out things

File: src/config/fabric-ansible/ansible-playbooks/group_vars/all.yml [hunk 4]
```diff
@@ -174,3 +174,31 @@ TFTP_BOOT_PATH: "/var/lib/docker/volumes/ironic_pxe/_data"
...
+
+contrail_command_node: 
+  auth_host: 1.1.1.1
+  username: admin
+  password: admin
+
+ironic_node:
+  auth_url: http://10.10.10.10:5000/v3 
+  username: admin
+  password: 905bee05b9f8dda6c3f5eee0aed8056e214b77f6
```
Comment (nitishkrishna) [inline src/config/fabric-ansible/ansible-playbooks/group_vars/all.yml]: don't set passwords in GitHub uploaded files

File: src/config/fabric-ansible/ansible-playbooks/server_discovery.yml [hunk 5]
```diff
@@ -0,0 +1,41 @@
+---
```
Comment (nitishkrishna) [inline src/config/fabric-ansible/ansible-playbooks/server_discovery.yml]: split file into three task files:
1. IPMI
2. Plugin related tasks
3. CC related tasks (if needed)

**Comment by pcarver** [comment]
Contrail/Tungsten Fabric does not accept GitHub pull requests. All changes should be submitted via the Gerrit code review tool. If you're not familiar with Gerrit, take a look at section 4 of https://github.com/tungstenfabric/docs/blob/master/Contributor/GettingStarted/getting-started-with-opencontrail-development.md for more info.

The key points are that you need to login to https://review.opencontrail.org with an Ubuntu One account in order to establish your SSH public key. You need to clone the repos from Gerrit rather than (or in addition to) GitHub (or manually add an additional remote to an existing repo rather than cloning if you prefer). And you need to install and use the "git review" extension to Git in order to push changes to Gerrit.

**Pull request closed without merging**